experimental/cmd/pca9548/main.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// pca9548 scans the 8 ports of a pca9548 i2c multiplexer for other i2c devices.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/experimental/devices/pca9548"
+	"periph.io/x/periph/host"
+)
+
+func mainImpl() error {
+	address := flag.Int("address", 0x70, "I²C address")
+	i2cbus := flag.String("bus", "", "I²C bus (/dev/i2c-1)")
+
+	flag.Parse()
+
+	if _, err := host.Init(); err != nil {
+		return err
+	}
+
+	// Open default I²C bus.
+	bus, err := i2creg.Open(*i2cbus)
+	if err != nil {
+		return fmt.Errorf("failed to open I²C: %v", err)
+	}
+	defer bus.Close()
+
+	// Registers a multiplexer with 8 ports at address 0x70 if no other address
+	// supplied with command line option.
+	_, err = pca9548.Register(bus, &pca9548.Opts{Address: *address})
+	if err != nil {
+		return fmt.Errorf("failed to load new mux: %v", err)
+	}
+
+	devices := make(map[string][]uint16)
+	fmt.Println("Starting Scan")
+	for i := 0; i < 8; i++ {
```

Comment:
Could you not hard code it, could the number of ports exposed as a method on Dev? This would open the door to support different chip variation.

Reply:
I'd still prefer `8` to not be hardcoded, and instead retrieved from mux.